### PR TITLE
fix: log warning for managers which don't support replacements

### PR DESCRIPTION
**Pull request by JamieMagee**
## Changes:

Add a log message for the managers which don't support replacements yet.

## Context:

https://github.com/renovatebot/renovate/pull/13211#issuecomment-1000171457

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

**Review by viceice — DISMISSED** [review]

**Comment by viceice** [comment]
I think this should be a fix PR?

File: lib/manager/maven/update.ts [hunk 1]
```diff
@@ -28,6 +28,10 @@ export function updateDependency({
   fileContent,
   upgrade,
 }: UpdateDependencyConfig): string | null {
+  if (upgrade.updateType === 'replacement') {
+    logger.warn('gomod manager does not support replacement updates yet');
```
Comment (astellingwerf) [inline lib/manager/maven/update.ts:32]:
```suggestion
    logger.warn('maven manager does not support replacement updates yet');
```

File: lib/manager/gradle/shallow/update.ts [hunk 2]
```diff
@@ -7,7 +7,11 @@ export function updateDependency({
   fileContent,
   upgrade,
 }: UpdateDependencyConfig<GradleManagerData>): string | null {
-  const { depName, currentValue, newValue, managerData } = upgrade;
+  const { depName, currentValue, newValue, managerData, updateType } = upgrade;
+  if (updateType === 'replacement') {
+    logger.warn('maven manager does not support replacement updates yet');
```
Comment (astellingwerf) [inline lib/manager/gradle/shallow/update.ts:12]:
```suggestion
    logger.warn('gradle manager does not support replacement updates yet');
```

**Review by astellingwerf — CHANGES_REQUESTED** [review]
Two of the managers warnings are very confusing.

**Comment by JamieMagee** [comment]
> Two of the managers warnings are very confusing.

Thanks! Copy-paste error 🤦

**Comment by JamieMagee** [comment]
> I think this should be a fix PR?

Changed from `chore` to `fix`

**Review by rarkins — DISMISSED** [review]

**Review by viceice — COMMENTED** [review]
Looks like there's something wrong with the tests. Also some coverage seems missing.

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**